rust_src/src/windows.rs

```diff
@@ -1058,6 +1058,18 @@ pub fn scroll_down(arg: LispObject) {
     unsafe { scroll_command(arg, -1) };
 }
 
+/// Return new normal size of window WINDOW.
+/// WINDOW must be a valid window and defaults to the selected one.
+///
+/// The new normal size of WINDOW is the value set by the last call of
+/// `set-window-new-normal' for WINDOW.  If valid, it will be shortly
+/// installed as WINDOW's normal size (see `window-normal-size').
+#[lisp_fn(min = "0")]
+pub fn window_new_normal(window: LispWindowValidOrSelected) -> LispObject {
+    let win: LispWindowRef = window.into();
+    win.new_normal
```

Comment:
When we port to Rust, we try to return as specific a type as possible. From my reading of the code, it looks like `new_normal` is a floating type. Perhaps this should return `EmacsDouble` and the return should be `win.new_normal.as_float_or_error()`? Or are there cases where `nil` is valid? If so, the return can be `Option<EmacsDouble>` and the code can return `win.new_normal.as_float()`.

The reason for the specific return is it allows for the Rust code to act on native types instead of marshalling to and from Lisp primitives.

Reply:
Ahh, a very good point. 

As best as I can tell, there are no cases where nil is valid -- but to be honest, I'm new here, and I'd welcome advice on this. To my read, returning `EmacsDouble` sounds ideal.

Reply:
Let's start out cautious and return `Option<EmacsDouble>`. We can switch to plain `EmacsDouble` as the port continues.

Reply:
It appears `window-new-normal` can return any `LispObject`. `(set-window-new-normal nil "abc") (window-new-normal)` returns `"abc"`.

I thought this is a mistake in restricting the value but see `lisp/window.el` line 5351 -

`      (unless (eq (window-new-normal sub) 'ignore)`

This suggests that `window-new-normal` returning a symbol is an expected behaviour.

Reply:
I suspect the assignment of a string is lazy checking. However, it does look like a symbol is checked. Which means the return should remain LispObject.

The tests should reflect the checking against a symbol. The string is a mistake we should perhaps prevent at some point.

Reply:
Okay. I'll move this back to `LispObject`, perhaps with a `TODO` comment to investigate and refine the type as more of `window.c` is ported.